backbone/src/main/java/org/researchstack/backbone/ui/step/layout/ProfileStepLayout.java

```diff
@@ -0,0 +1,238 @@
+ *        which can be found in the @see {@link org.researchstack.backbone.step.ProfileStep}
+ */
+
+public class ProfileStepLayout extends FormStepLayout {
+
+    protected User user;
+
+    /** Used to map steps identifiers to error messages */
+    protected Map<String, String> identifierErrorMap;
+
+    public ProfileStepLayout(Context context) {
+        super(context);
+    }
+
+    public ProfileStepLayout(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public ProfileStepLayout(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    public ProfileStepLayout(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
+        super(context, attrs, defStyleAttr, defStyleRes);
+    }
+
+    protected ProfileStep getProfileStep() {
+        if (!(formStep instanceof ProfileStep)) {
+            throw new IllegalStateException("ProfileStepLayout must contain a ProfileStep");
+        }
+        return (ProfileStep)formStep;
+    }
+
+    @Override
+    public void initialize(Step step, StepResult result, TaskResult taskResult) {
+        validateStep(step);  // also sets formStep variable
+        initializeErrorMap();
+        prePopulateUserProfileResults(step, result);
+        super.initialize(step, result, taskResult);
+    }
+
+    protected void initializeErrorMap() {
+        identifierErrorMap = new HashMap<>();
+        identifierErrorMap.put(ProfileInfoOption.NAME.getIdentifier(), getString(R.string.rsb_error_invalid_name));
+        identifierErrorMap.put(ProfileInfoOption.EMAIL.getIdentifier(), getString(R.string.rsb_error_invalid_email));
+        identifierErrorMap.put(ProfileInfoOption.PASSWORD.getIdentifier(), getString(R.string.rsb_error_invalid_password));
+        // TODO: add the rest of the user profile error messages for options
+    }
+
+    /**
+     * @param result to add pre-populated user profile results that are create from the User object
+     */
+    protected void prePopulateUserProfileResults(Step step, StepResult result) {
+        user = DataProvider.getInstance().getUser(getContext());
+        if (user == null) {
+            user = new User();  // first time controlling the user object
+        }
+        if (result == null) {
+            result = new StepResult<StepResult>(step);
+        }
+        for (ProfileInfoOption option : getProfileStep().getProfileInfoOptions()) {
+            // Look to see if the step result for this profile option already exists
+            StepResult profileResult = StepResultHelper.findStepResult(result, option.getIdentifier());
+            // If it doesn't exist, create one matching the profile info option type from User object
+            if (profileResult == null) {
+                Step profileStep = StepHelper.getStepWithIdentifier(formStep.getFormSteps(), option.getIdentifier());
```

Comment:
A FormStep displays multiple QuestionSteps in one StepLayout screen.  For instance, Eligibility Quiz is a FormStep. Tasks are meant to simply control the movement from step to step, so if you put a list of steps in a Task, it would put each QuestionStep on their own screen, where a FormStep needs multiple steps on one screen.